waiter/src/waiter/scheduler.clj

```diff
@@ -665,6 +666,40 @@
                         service
                         (assoc instances :active-instances active-instances))))))))
 
+(defn start-new-services-handler
+  "Listens on token-metric-chan-mult and determines whether to trigger start new services for a token. Only the leader
+  router starts new services. Tokens with new last-request-time pointing to a service descriptor that does not exist will
+  have the latest service started.
+
+  Returns a map with keys:
+  :process-token-event-ch is a channel where messages are pushed to it when a service was started for a token event."
+  [retrieve-latest-descriptor-fn kv-store token-metric-chan-mult start-new-service-fn leader?-fn fallback-state-atom]
```

Comment:
Can we pass in a curried `service-exists?` function, then we do not pass in fallback state atom here? Then we to worry about the atom being changed by `start-new-services-handler` function.